taichi/backends/opengl/opengl_api.cpp

```diff
@@ -277,8 +289,16 @@ void initialize_opengl() {
     TI_TRACE("[glsl] Found " #x);
 #include "taichi/inc/opengl_extension.inc.h"
 #undef PER_OPENGL_EXTENSION
-  if (!opengl_has_GL_ARB_compute_shader)
+  if (!opengl_has_GL_ARB_compute_shader) {
+    if (noerr) {
+      TI_INFO("Your OpenGL does not support GL_ARB_compute_shader extension");
+      return false;
+    }
     TI_ERROR("Your OpenGL does not support GL_ARB_compute_shader extension");
+  }
+
+  gl_supp = true;
```

Comment:
Should here be `gl_init`?

Reply:
Let's use `std::optional` to make the logic here better.